thermal-control-design.md

```diff
@@ -0,0 +1,272 @@
+# SONiC Thermal Control Design #
+
+### Rev 0.1 ###
+
+### Revision ###
+
+ | Rev |     Date    |       Author       | Change Description                |
+ |:---:|:-----------:|:------------------:|-----------------------------------|
+ | 0.1 |             |      Liu Kebo      | Initial version                   |
+ | 0.2 |             |      Liu Kebo      | Revised after community review    |
+
+
+  
+## 1. Overview
+
+The purpose of Thermal Control is to keep the switch at a proper temperature by using cooling devices, e.g., fan.
+Thermal control daemon need to monitor the temperature of devices (CPU, ASIC, optical modules, etc) and the running status of fan. It store temperature values fetched from sensors and thermal device running status to the DB, to make these data available to CLI and SNMP or other apps which interested. 
+
+Thermal control also enforce some environment related polices to help the thermal control algorithm to adjust the switch temperature.     
+
+## 2. Thermal device monitoring
+
+Thermal monitoring function will retrieve the switch device temperatures via platform APIs, platform APIs not only provide the value of the temperature, but also provide the threshold value. The thermal object status can be deduced by comparing the current temperature value against the threshold. If it above the high threshold or under the low threshold, alarm shall be raised.
```

Comment:
Is the thermal/PSU/FAN monitoring daemon going to run as a separate entity than the thermal control daemon described in 3.1 ? If so, please indicate if the polling frequency of the monitoring daemon can be configurable ?

It would very much be desirable to run the monitoring/poller daemon separately (especially in BMC based units) so the thermals/PSU/FAN status may be used for CLI/diagnostics/telemetry purposes (since BMC would handle the thermal control).

Reply:
the monitor functions and thermal control part will be two separated threads of one daemon.  we will provide the flexibility for the vendors to decide not to run the thermal control part if they decide to cover this part somewhere else, in your case by the BMC, so this should be not an issue. 

for the polling interval,  it's not configurable from the CLI, because currently we don't have a framework to support this is kind of configuration for pmon.

Reply:
Thanks for providing flexibility to not run thermal control part.
w.r.t polling interval, will it be configurable using a configDB entry if not CLI? since we may not want to enforce the same interval across all platforms..

Reply:
also not possible to config from config DB under curent pmon framework, can consider to add it as part of the thermal control daemon config file(we have a file to config the policies)

Reply:
Any way to configure this per device during init would be good.